utils/difference.js

```diff
@@ -48,6 +50,38 @@ function calculateDifferences(row) {
   }
 }
 
+function calculatePreviousDifferences(row) {
+  const {
+    previous_sum,
+    previous_comparison_sum,
+    previous_m,
+    previous_comparison_m,
+  } = row;
+
+  const isDecennial = row.profile === 'decennial';
+  const hasValidInputs = allExist(previous_sum, previous_comparison_sum, previous_m, previous_comparison_m)
+  const shouldNullify = (!hasValidInputs
+    || row.previous_codingThreshold // TODO: why is this here?
+    || row.previous_comparison_codingThreshold // TODO: why is this here?
+  ) && !isDecennial;
+
+  if (shouldNullify) {
+    nullPreviousDifferences(row);
+  } else {
+    row.previous_difference_sum = executeFormula('delta', [row.previous_sum, row.previous_comparison_sum]);
```

Comment:
the old code is so weird/crazy, it's like why do we create these constants in the beginning if we are not using them later on?
```javascript
    row.previous_difference_sum = executeFormula('delta', [previous_sum, previous_comparison_sum]);
```